cashctrl_ledger/ledger.py

```diff
@@ -265,13 +266,21 @@ def _single_account_balance(
         response = self._client.request("GET", "account/balance", params=params)
         balance = float(response.text)
 
+        account_df = self.accounts.list().query("account == @account")
+        is_negate = account_df['group'].apply(
+            lambda x: any(
+                x.startswith(f"/{category}")
+                for category in ACCOUNT_CATEGORIES_NEED_TO_NEGATE
+            )
+        ).any()
```

Comment:
How about something along the lines of:
```
is_naegate = pd.Series(FALSE*len(account_df))
for category in ACCOUNT_CATEGORIES_NEED_TO_NEGATE:
    is_negate = is_negate  | account_df['group'].str.statrswith(f"/{category}")
```
instead of:
```
is_negate = account_df['group'].apply(
    lambda x: any(
        x.startswith(f"/{category}")
        for category in ACCOUNT_CATEGORIES_NEED_TO_NEGATE
    )
).any()
```

Reply:
I haven't realized that account is a singular when writing above suggestion.
Since account is a singular, I suggest an even more readable version:
```
group = self.accounts.list().query("account == @account")["group"].item()
root_category = re.sub("/.*", "", re.sub("^/", "", group))
if root_category in ACCOUNT_CATEGORIES_NEED_TO_NEGATE:
    balance = balance * -1
```
I have force-pushed to the branch to  remove commit f947784 (logic according to my initial comment) and willl  push 99eea7f instead with the newly proposed logic.